lib/backend/model/block.go

```diff
@@ -31,7 +32,8 @@ var (
 )
 
 type BlockKey struct {
-	CIDR net.IPNet `json:"-" validate:"required,name"`
+	CIDR net.IPNet  `json:"-" validate:"required,name"`
+	UID  *types.UID `json:"-"`
```

Comment:
Should this be part of the key or should it be part of the KVPair?

Reply:
Good question - I was originally imagining this as part of the identifier for a resource, which is why I put it in the Key.

But, I guess it's analogous to the revision, which isn't included in the key. I could see it going either way.

Reply:
Ah, the other issue is that `Delete` doesn't take a KVPair - it just takes a Key.

I guess we could update everything to pass around KVPs without Values, even for delete?

Reply:
Heh, even the comment on Delete says it takes a KVPair :-)  

Maybe add a `DeleteKVPair()` method that takes revision/UID from the KVP instead of changing all callers?  I do feel like the UID is more akin to the revision than the key.  Adding it to the key breaks assumptions about the key being 1-1 with the name of the object, which might have unintended consequences.

Reply:
Yep, that sounds good to me. Fixed in latest commit.